lib/metadata_json_lint.rb

```diff
@@ -136,10 +142,26 @@ def validate_requirements!(requirements)
       if requirement['name'] == 'pe'
         warn :requirements, "The 'pe' requirement is no longer supported by the Forge."
       end
+
+      begin
+        VersionRequirement.new(requirement.fetch('version_requirement', ''))
+      rescue ArgumentError => e
+        # Raised when the version_requirement provided could not be parsed
+        error :dependencies, "Invalid 'version_requirement' field in metadata.json: #{e}"
+      end
+
+      validate_puppet_ver!(requirement) if options[:strict_puppet_version]
     end
   end
   module_function :validate_requirements!
 
+  def validate_puppet_ver!(range)
+    min_ver = range['version_requirement'].gsub(/[><=~^]/, '').split(' ')[0]
+
+    warn(:requirements, "#{min_ver} is no longer a supported version of Puppet. Minimum supported version is 4.10.0.") if min_ver =~ /^[2-4]\.[0-9]\.\d$/
```

Comment:
`3.x` is semantically valid but you'd miss it with the current code. Same for `>= 3.0` or if anyone used `3.10` instead of `4.10`

Reply:
I'll take a look at this and fix it.

Reply:
@ekohl scratch that. I am going to update the code to use the VersionRequirement class instead of a bunch of duplicate inline code, but we already warn on the use of things like `3.x` and `3.*`, which by default will cause a failure as `--fail-on-warnings` is turned on by default,